### PR TITLE
Reading V4 Token

### PR DESCRIPTION
**Pull request by Egge21M**
# Fixes: None

## Description
This PR changes the way `handleTokens` works. It also adds reading the newly proposed v4 Token format. V4 token are parsed and mapped to output the exact same structure as v3 Token to make handling easier. Consumers of `getDecodedToken` receive a stable result.

## Changes

- Removed deprecated token format support (pre v3)
- Removed test cases for deprecated token formats
- `handleTokens` now parses the version and decodes accordingly.

## PR Tasks

- [x] Open PR
- [x] Refactoring and remove logs
- [x] run `npm run test` --> no failing unit tests
- [ ] run `npm run format`

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -97,21 +98,33 @@ function getDecodedToken(token: string): Token {
...
-
-	// check if v3
-	if ('token' in obj) {
-		return obj;
-	}
-
-	// check if v1
-	if (Array.isArray(obj)) {
-		return { token: [{ proofs: obj, mint: '' }] };
+function handleTokens(token: string): Token | undefined {
```
[the remark this answers is not in the document]
Comment (Egge21M) [reply src/utils.ts]: Good catch. Fixed

**Review by gudnuf — APPROVED** [review]
lgtm! 233b8baef31c4f3f79f13801b542a5c358cb4bf1
Thanks for adding cbor directly to this lib, I've had issues in the past with cbor and server-side rendering, but just tested in a next.js app and works great!

**Comment by Egge21M** [comment]
> lgtm! [233b8ba](https://github.com/cashubtc/cashu-ts/commit/233b8baef31c4f3f79f13801b542a5c358cb4bf1) Thanks for adding cbor directly to this lib, I've had issues in the past with cbor and server-side rendering, but just tested in a next.js app and works great!

Great to hear that! Please be aware that this implementation is very limited in scope and was only tested against token v4 specific cases. There might be edge cases that are not covered

**Comment by Egge21M** [comment]
FYI this is rebased to the new dev branch and ready for a release in 1.X

**Comment by callebtc** [comment]
Amazing work, especially implementing the cbor de/serializer.

**Review by callebtc — APPROVED** [review]
ut-LGTM

**Pull request merged**